modules/base/manifests/init.pp

```diff
@@ -2,24 +2,38 @@
 #
 #
 
-class base {
+class base (
+	$period 	 	 = 'weekly',
+	$repeat 	 	 = '1'
+){
+	# schedule parameters for update
+	schedule { 'update':
```

Comment:
Read documentation about "schedule" carefully.

> Currently, schedules can only be used to stop a resource from being applied; they cannot cause a resource to be applied when it otherwise wouldn’t be, and they cannot accurately specify a time when a resource should run.


I would not use it. But the last decision is up to you.